pkg/cableengine/healthchecker/pinger.go

```diff
@@ -0,0 +1,78 @@
+package healthchecker
+
+import (
+	"time"
+
+	"github.com/go-ping/ping"
+	"k8s.io/klog"
+)
+
+var waitTime time.Duration = 15
+var size uint64 = 1000
+
+type Pinger struct {
+	healthCheckIp string
+	statistics    Statistics
+	Status        string
+	stopCh        chan struct{}
+	pinger        *ping.Pinger
+}
+
+func NewPinger(healthCheckIP string) *Pinger {
+	return &Pinger{
+		healthCheckIp: healthCheckIP,
+		statistics: Statistics{
+			size:         size,
+			previousRtts: make([]uint64, size),
+		},
+		stopCh: make(chan struct{}),
+	}
+}
+
+func (p *Pinger) Run() {
+	go func() {
+		var err error
+
+		for {
+			select {
+			case <-p.stopCh:
+				return
+			default:
+				// Wait for a while for the connection to be ready.
+				time.Sleep(waitTime * time.Second)
+
+				p.pinger, err = ping.NewPinger(p.healthCheckIp)
+				if err != nil {
+					klog.Errorf("Creating pinger for the ip %v failed due to %v", p.healthCheckIp, err)
+
+					return
+				}
+
+				p.sendPing(p.pinger, p.healthCheckIp)
+			}
+		}
+	}()
+	klog.Infof("CableEngine HealthChecker started")
+}
+
+func (p *Pinger) stop() {
+	p.pinger.Stop()
+}
+
+func (p *Pinger) sendPing(pinger *ping.Pinger, healthCheckIp string) {
+	pinger.SetPrivileged(true)
+	pinger.RecordRtts = false
+
+	pinger.OnRecv = func(packet *ping.Packet) {
+		p.Status = ""
+		p.statistics.updateStatistics(uint64(packet.Rtt.Nanoseconds()))
+	}
+
+	pinger.OnFinish = func(stats *ping.Statistics) {
+		p.Status = "Failed to successfully ping the remote endpoint"
+	}
```

Comment:
The docs say this is called when the Pinger exits so doesn't seem indicative of an error. It seems we can remove this.

Reply:
since we are not setting the count, it will be an endless ping. ff there are no successful packets till  the Timeout the onFinish will be called and it is a failed ping.

The timeout setting seems to be missed during the patch rework I guess I have added it back.